src/wallet/wallet.ts

```diff
@@ -103,170 +142,442 @@ class HathorWalletServiceWallet extends EventEmitter {
         if (data && data.error === 'wallet-already-loaded') {
           // If it was already loaded, we have to check if it's ready
           await handleCreate(data.status);
+        } else {
+          throw new WalletRequestError('Error creating wallet.');
         }
+      } else {
+        throw new WalletRequestError('Error creating wallet.');
       }
-      // TODO Should we handle another error?
     }
   }
 
+  /**
+   * Return wallet auth token
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  getAuthToken(): string | null {
+    return this.authToken;
+  }
+
   /**
    * When the wallet starts, it might take some seconds for the wallet service to completely load all addresses
    * This method is responsible for polling the wallet status until it's ready
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
   async startPollingStatus() {
     try {
-      const res = await walletApi.getWalletStatus(this.walletId!);
+      const res = await walletApi.getWalletStatus(this);
       const data = res.data;
```

Comment:
Shouldn't we move this logic to the `walletApi`? I think this code must be way simpler to read because it would have to deal with a promise that might succeed or fail.

Reply:
Makes sense, done.

Reply:
This refactor is great. It is way easier to read the code. Thanks!